src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -320,6 +320,22 @@ public static ISettings LoadSpecificSettings(string root, string configFileName)
                 useTestingGlobalPath: false);
         }
 
+        public static ISettings LoadImmutableSettingsGivenConfigPaths(IList<string> configFilePaths, SettingsLoadingContext settingsLoadingContext)
+        {
+            var settings = new List<SettingsFile>();
+            if (configFilePaths == null || configFilePaths.Count == 0)
+            {
+                return NullSettings.Instance;
+            }
+
+            foreach (var configFile in configFilePaths)
```

Comment:
`configFile` => `configFilePath` because it's a file path not a file.

Then you can remove [this named parameter](https://github.com/NuGet/NuGet.Client/pull/3086/files#diff-86990f2adab67354828de3f4f08b388cR333).